src/effects/effectslot.cpp

```diff
@@ -150,72 +299,83 @@ EffectButtonParameterSlotPointer EffectSlot::getEffectButtonParameterSlot(unsign
     return m_buttonParameters[slotNumber];
 }
 
-void EffectSlot::loadEffect(EffectPointer pEffect, bool adoptMetaknobPosition) {
-    //qDebug() << debugString() << "loadEffect"
-    //         << (pEffect ? pEffect->getManifest().name() : "(null)");
-    if (pEffect) {
-        m_pEffect = pEffect;
-        m_pControlLoaded->forceSet(1.0);
-        m_pControlNumParameters->forceSet(pEffect->numKnobParameters());
-        m_pControlNumButtonParameters->forceSet(pEffect->numButtonParameters());
+void EffectSlot::loadEffect(EffectManifestPointer pManifest, EffectInstantiatorPointer pInstantiator,
+        const QSet<ChannelHandleAndGroup>& activeChannels) {
+    // Remove an effect if already loaded in slot
+    if (m_pEngineEffect != nullptr) {
+        unloadEffect();
+    }
 
-        // The enabled status persists in the EffectSlot when loading a new
-        // EffectPointer to the EffectSlot. Effects and EngineEffects default to
-        // disabled, so if this EffectSlot was enabled, enable the Effect and EngineEffect.
-        pEffect->setEnabled(m_pControlEnabled->toBool());
+    if (pManifest != nullptr || pInstantiator != nullptr) {
```

Comment:
Add a comment: "In this case, only unload the old effect and return"

Reply:
This logic is backwards. Change to:
`if (pManifest == nullptr || pInstantiator == nullptr) {`

Reply:
also move `m_pManifest = pManifest` above this